Rel notes for storage 4.18 (#86629)
openshift/openshift-docs: release_notes/ocp-4-18-release-notes.adoc

[id="ocp-4-18-storage-editing-overprovisioning-ratio_{context}"]
==== Over-provisioning ratio update after LVMCluster custom resource creation
Previously, the `thinPoolConfig.overprovisionRatio` field in the `LVMCluster` custom resource (CR) could be configured only during the creation of the `LVMCluster` CR. With this release, you can now update the `thinPoolConfig.overprovisionRatio` field even after creating the `LVMCluster` CR.

> [id="ocp-4-18-release-notes-storage-cifs-smb-csi-driver_{context}"]

Reviewer: LGTM for smb csi, we already add the limitations use cases and remove the TP in https://github.com/openshift/openshift-docs/pull/84302

Reviewer 2: +1. LGTM for smb csi part.

==== Persistent storage using CIFS/SMB CSI Driver Operator is generally available
{product-title} is capable of provisioning persistent volumes (PVs) with a Container Storage Interface (CSI) driver for the Common Internet File System (CIFS) dialect/Server Message Block (SMB) protocol. The CIFS/SMB CSI Driver Operator that manages this driver was introduced in {product-title} 4.16 with Technology Preview status. In {product-title} 4.18, it is now generally available.

For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-smb-cifs.adoc#persistent-storage-csi-smb-cifs[CIFS/SMB CSI Driver Operator].

[id="ocp-4-18-release-notes-storage-secret-csi-driver_{context}"]

> ==== Secret Store CSI Driver Operator is generally available

Reviewer 3: Content looks good to me for Secret Store part. 
Double confirm "to mount multiple secrets, keys, and certificates stored" => "to mount multiple secrets, keys and certificates stored" would be better?

Author: We generally use the serial (aka Oxford comma).

> The Secrets Store Container Storage Interface (CSI) Driver Operator, `secrets-store.csi.k8s.io,` allows {product-title} to mount multiple secrets, keys, and certificates stored in enterprise-grade external secrets stores into pods as an inline ephemeral volume. The Secrets Store CSI Driver Operator communicates with the provider using gRPC to fetch the mount contents from the specified external secrets store. After the volume is attached, the data in it is mounted into the container’s file system. The Secrets Store CSI Driver Operator was available in {product-title} 4.14 as a Technology Preview feature. {product-title} 4.18 introduces this feature as generally available. For more information about the Secrets Store CSI driver, see xref:../storage/container_storage_interface/persistent-storage-csi-secrets-store.adoc#persistent-storage-csi-secrets-store[Secrets Store CSI Driver Operator].

Reviewer 4: ```suggestion
The Secrets Store Container Storage Interface (CSI) Driver Operator, `secrets-store.csi.k8s.io`, allows {product-title} to mount multiple secrets, keys, and certificates stored in enterprise-grade external secrets stores into pods as an inline ephemeral volume. The Secrets Store CSI Driver Operator communicates with the provider using gRPC to fetch the mount contents from the specified external secrets store. After the volume is attached, the data in it is mounted into the container’s file system. The Secrets Store CSI Driver Operator was available in {product-title} 4.14 as a Technology Preview feature. {product-title} 4.18 introduces this feature as generally available. For more information about the Secrets Store CSI driver, see xref:../storage/container_storage_interface/persistent-storage-csi-secrets-store.adoc#persistent-storage-csi-secrets-store[Secrets Store CSI Driver Operator].
```

-  A tiny nit to adjust the comma. 
-  You can consider dividing this into paragraphs.

// For information about using the Secrets Store CSI Driver Operator to mount secrets from an external secrets store to a CSI volume, see xref:../nodes/index#nodes-pods-secrets-store[Providing sensitive data to pods by using an external secrets store].

> [id="ocp-4-18-release-notes-storage-pv-last-phase-transition-time_{context}"]

Reviewer 5: @lpettyjo lgtm

==== Persistent volume last phase transition time parameter is generally available

> {product-title} 4.16 introduced a new parameter, `LastPhaseTransitionTime`, which has a timestamp that is updated every time a persistent volume (PV) transitions to a different phase (`pv.Status.Phase`). For {product-title} 4.18, this feature is generally available.

Reviewer 6: Content LGTM

> For more information about using the persistent volume last phase transition time parameter, see xref:../storage/understanding-persistent-storage.adoc#persistent-volumes_understanding-persistent-storage[Persistent volumes].

Reviewer 6: This reference takes to PV docs, but doesn't have details about "last phase transition time parameter"

Author: @radeore Correct. Since the PR for the "last phase transition time parameter" hasn't merged yet, you can't see the content, but the link is to the correct file.

[id="ocp-4-18-release-notes-storage-volume-multi-vcenter-support_{context}"]
==== Multiple vCenter support for vSphere CSI is generally available
{product-title} 4.17 introduced the ability to deploy {product-title} across multiple vSphere clusters (vCenters) as a Technology Preview feature. In {product-title} 4.18, Multiple vCenter support is now generally available.

> For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-vsphere.adoc#persistent-storage-csi-vsphere-multi-vcenter-support-overview_persistent-storage-csi-vsphere[Multiple vCenter support for vSphere CSI] and xref:../installing/installing_vsphere/installation-config-parameters-vsphere.adoc[Installation configuration parameters for vSphere].

Reviewer 7: It's not the scope of this RN PR,  but let's ensure we update the reference `Multiple vCenter support for vSphere CSI` to remove the TP.

Author: @duanwei33 Yes. That is covered in this PR https://github.com/openshift/openshift-docs/pull/85971.

[id="ocp-4-18-release-notes-storage-always-honor-reclaim-policy_{context}"]

> ==== Always apply persistent volume reclaim policy (Technical Preview)

Reviewer 6: ```suggestion
==== Always honor persistent volume reclaim policy (Technical Preview)
```

Reviewer 6: In case to keep title consistent with Upstream feature

Author: Done.

Prior to {product-title} 4.18, the persistent volume (PV) reclaim policy was not always honored.

> For a bound PV-persistent volume pair (PVC) pair, the ordering of PV-PVC deletion determined whether the PV delete reclaim policy was applied or not. The PV applied the reclaim policy if the PVC was deleted prior to deleting the PV. However, if the PV was deleted prior to deleting the PVC, then the reclaim policy was not applied. As a result of that behavior, the associated storage asset in the external infrastructure was not removed.

Reviewer 4: ```suggestion
For a bound PV-persistent volume claim (PVC) pair, the ordering of PV-PVC deletion determined whether the PV delete reclaim policy was applied or not. The PV applied the reclaim policy if the PVC was deleted prior to deleting the PV. However, if the PV was deleted prior to deleting the PVC, then the reclaim policy was not applied. As a result of that behavior, the associated storage asset in the external infrastructure was not removed.
```

I believe it should be "claim"

> For a bound PV-persistent volume claim (PVC) pair, the ordering of PV-PVC deletion determined whether the PV delete reclaim policy was applied or not. The PV applied the reclaim policy if the PVC was deleted prior to deleting the PV. However, if the PV was deleted prior to deleting the PVC, then the reclaim policy was not applied. As a result of that behavior, the associated storage asset in the external infrastructure was not removed.

Reviewer 6: ```suggestion
For a bound persistent volume (PV)-persistent volume claim (PVC) pair, the ordering of PV-PVC deletion determined whether the PV delete reclaim policy was applied or not. The PV applied the reclaim policy if the PVC was deleted prior to deleting the PV. However, if the PV was deleted prior to deleting the PVC, then the reclaim policy was not applied. As a result of that behavior, the associated storage asset in the external infrastructure was not removed.
```

Author: @radeore It's awkward, but the PV acronym is define in the sentence above this, but PVC is not, so it needs to be defined here. I've removed the hyphen to make this a little less jarring (I hope).

> With {product-title} 4.18, the PV reclaim policy is consistently always applied. This feature has Technical Preview status.

Reviewer 6: Should we have this feature listed under "Storage TechPreview features" table? cc. @duanwei33

Author: @radeore It's technically a feature, but it seems more to be addressing something that should have behaved this way all along, so I didn't list it. @gcharot @jsafrane Do you want me to list this in the features table?

Author: Leaving out of table.

For more information, see xref:../storage/understanding-persistent-storage.adoc#reclaiming_understanding-persistent-storage[Reclaim policy for persistent volumes].

[id="ocp-4-18-release-notes-storage-lso-cleanup_{context}"]
==== Improved ability to easily remove LVs or LVSs is generally available
{product-title} 4.18 improves the ability to remove Local Volumes (LVs) and Local Volume Sets (LVSs) by automatically removing artifacts, thus reducing the number of steps required.

For more information, see xref:../storage/persistent_storage/persistent_storage_local/persistent-storage-local.adoc#local-removing-device_persistent-storage-local[Removing a local volume or local volume set].

[id="ocp-4-18-release-notes-storage-volume-group-snapshots_{context}"]

> ==== CSI volume group snapshots is generally available

Reviewer 7: The content looks good, but please note the code is still not ready now and we need to recheck this part before release.

Reviewer 8: correct, it may slip GA and added in zstream

Reviewer 8: @lpettyjo final decision, this goes tech preview

Author: Done.

{product-title} 4.18 introduces Container Storage Interface (CSI) volume group snapshots as a generally available feature. CSI volume group snapshots use a label selector to group multiple persistent volume claims (PVCs) for snapshotting. A volume group snapshot represents copies from multiple volumes that are taken at the same point-in-time. This can be useful for applications that contain multiple volumes. 

{product-title} supports volume group snapshots by default. However, the only storage type in {product-title} that supports volume group snapshots is {rh-storage}.

//For more information about CSI volume group snapshots, see xref:/../storage/container_storage_interface/persistent-storage-csi-group-snapshots.adoc#persistent-storage-csi-group-snapshots[CSI volume group snapshots].

[id="ocp-4-18-release-notes-storage-c3-n4-instance-types-hyperdisk_{context}"]
==== GCP PD CSI driver supports the C3 instance type for bare metal and N4 machine series is generally available

> The Google Cloud Platform Persistent Disk(GCP PD) Container Storage Interface (CSI) driver supports the C3 instance type for bare metal and N4 machine series. The C3 instance type and N4 machine series support the hyperdisk-balanced disks.

Reviewer 4: ```suggestion
The Google Cloud Platform Persistent Disk (GCP PD) Container Storage Interface (CSI) driver supports the C3 instance type for bare metal and N4 machine series. The C3 instance type and N4 machine series support the hyperdisk-balanced disks.
```

[id="ocp-4-18-release-notes-storage-always-honor-reclaim-policy_{context}"]
==== Always honored persistent volume reclaim policy (Technical Preview)
Prior to {product-title} 4.18, the persistent volume (PV) reclaim policy was not always applied.

For a bound PV and persistent volume claim (PVC) pair, the ordering of PV-PVC deletion determined whether the PV delete reclaim policy was applied or not. The PV applied the reclaim policy if the PVC was deleted prior to deleting the PV. However, if the PV was deleted prior to deleting the PVC, then the reclaim policy was not applied. As a result of that behavior, the associated storage asset in the external infrastructure was not removed.

With {product-title} 4.18, the PV reclaim policy is consistently always applied. This feature has Technical Preview status.

For more information, see xref:../storage/understanding-persistent-storage.adoc#reclaiming_understanding-persistent-storage[Reclaim policy for persistent volumes].

[id="ocp-4-18-release-notes-storage-lso-cleanup_{context}"]
==== Improved ability to easily remove LVs or LVSs is generally available

> {product-title} 4.18 improves the ability to remove Local Volumes (LVs) and Local Volume Sets (LVSs) by automatically removing artifacts, thus reducing the number of steps required.

Reviewer 8: I think we should mention context here (i.e LSO)

Author: Done.

For more information, see xref:../storage/persistent_storage/persistent_storage_local/persistent-storage-local.adoc#local-removing-device_persistent-storage-local[Removing a local volume or local volume set].

[id="ocp-4-18-release-notes-storage-volume-group-snapshots_{context}"]
==== CSI volume group snapshots is generally available
{product-title} 4.18 introduces Container Storage Interface (CSI) volume group snapshots as a generally available feature. CSI volume group snapshots use a label selector to group multiple persistent volume claims (PVCs) for snapshotting. A volume group snapshot represents copies from multiple volumes that are taken at the same point-in-time. This can be useful for applications that contain multiple volumes. 

> {product-title} supports volume group snapshots by default. However, the only storage type in {product-title} that supports volume group snapshots is {rh-storage}.

Reviewer 8: Explicitly mention that this feature is CSI driver dependent. Wording is correct but needs to be rewritten to make sure people understand that the driver needs to implement the feature.

Not sure we should say "only", we have see a case where a third party driver adds supports after 4.18GA. Maybe just say ODF supports it?

Author: Done.

With {product-title} 4.18, the PV reclaim policy is consistently always applied. This feature has Technical Preview status.

For more information, see xref:../storage/understanding-persistent-storage.adoc#reclaiming_understanding-persistent-storage[Reclaim policy for persistent volumes].

[id="ocp-4-18-release-notes-storage-lso-cleanup_{context}"]
==== Improved ability to easily remove LVs or LVSs for LSO is generally available
For the Local Storage Operator (LSO), {product-title} 4.18 improves the ability to remove Local Volumes (LVs) and Local Volume Sets (LVSs) by automatically removing artifacts, thus reducing the number of steps required.

For more information, see xref:../storage/persistent_storage/persistent_storage_local/persistent-storage-local.adoc#local-removing-device_persistent-storage-local[Removing a local volume or local volume set].

[id="ocp-4-18-release-notes-storage-volume-group-snapshots_{context}"]
==== CSI volume group snapshots is generally available

> {product-title} 4.18 introduces Container Storage Interface (CSI) volume group snapshots as a generally available feature. The CSI Volume group snapshots feature is implemented by the CSI driver. CSI volume group snapshots use a label selector to group multiple persistent volume claims (PVCs) for snapshotting. A volume group snapshot represents copies from multiple volumes that are taken at the same point-in-time. This can be useful for applications that contain multiple volumes. 

Reviewer 8: > The CSI Volume group snapshots feature is implemented by the CSI driver

It's in OCP for core functionality AND ALSO needs to be supported by the driver. The wording may suggest that this feature entirely relies on the driver which is not the case. Maybe reword it with something like "The feature needs to be supported by the CSI driver"

Author: Done.

[id="ocp-4-18-release-notes-storage-lso-cleanup_{context}"]
==== Improved ability to easily remove LVs or LVSs for LSO is generally available
For the Local Storage Operator (LSO), {product-title} 4.18 improves the ability to remove Local Volumes (LVs) and Local Volume Sets (LVSs) by automatically removing artifacts, thus reducing the number of steps required.

For more information, see xref:../storage/persistent_storage/persistent_storage_local/persistent-storage-local.adoc#local-removing-device_persistent-storage-local[Removing a local volume or local volume set].

[id="ocp-4-18-release-notes-storage-volume-group-snapshots_{context}"]
==== CSI volume group snapshots (Technology Preview)
{product-title} 4.18 introduces Container Storage Interface (CSI) volume group snapshots as a Technology Preview feature. This feature needs to be supported by the CSI driver. CSI volume group snapshots use a label selector to group multiple persistent volume claims (PVCs) for snapshotting. A volume group snapshot represents copies from multiple volumes that are taken at the same point-in-time. This can be useful for applications that contain multiple volumes. 

> {product-title} supports volume group snapshots by default. {rh-storage} supports volume group snapshots.

Reviewer 8: I think we can remove this line because it's not enabled by default

Author: Done.

[id="ocp-4-18-removed-features_{context}"]
=== Removed features

[id="ocp-4-18-removals-storage_{context}"]
==== The Shared Resource CSI Driver is removed

> The Shared Resource CSI Driver feature is now removed from {product-title} 4.18. It was deprecated in {product-title} 4.17. This feature is now generally available in Builds for Red Hat OpenShift 1.1. To use this feature, ensure you are using Builds for Red Hat OpenShift 1.1 or later.

Reviewer: ```suggestion
The Shared Resource CSI Driver feature was deprecated in {product-title} 4.17 and is now removed from {product-title} 4.18. 
```
Maybe better.

Author: Done.

|{secrets-store-operator}
|Technology Preview
|Technology Preview
|General Availability

|CIFS/SMB CSI Driver Operator
|Technology Preview
|Technology Preview
|General Availability

|VMWare vSphere multiple vCenter support
|Not Available
|Technology Preview
|General Availability

|Disabling/enabling storage on vSphere
|Not Available
|Technology Preview
|Technology Preview

|RWX/RWO SELinux Mount
|Not Available
|Developer Preview
|Developer Preview

|Migrating CNS Volumes Between Datastores
|Not Available
|Developer Preview
|Developer Preview

|CSI volume group snapshots
|Not Available
|Not Available

> |General Availability

Reviewer 8: @lpettyjo we need TP here as well

Author: Done. Good catch!
